src/fluree/crypto/secp256k1.cljc

```diff
@@ -54,27 +55,40 @@ public key, hex encoded."
              :cljs (-> public .-y .toString (.replace #"^0x" "") encodings/pad-hex))]
     (encodings/x962-encode x y)))
 
+(defn- pad-to-length
+  "Left-pads str to length len with zeroes."
+  [str len]
+  (let [pad-len (- len (count str))]
+    (if (pos? pad-len)
+      (str/join (concat (repeat pad-len \0) str))
+      str)))
+
 (defn format-key-pair
   "Takes internal representation of a key-pair and returns X9.62 compressed encoded
   public key and private key as a map, with each value hex encoded."
   [pair]
-  (let [private #?(:clj (:private pair) :cljs (gobj/get pair "private"))
-        ^ECPoint public #?(:clj  (:public pair) :cljs (gobj/get pair "public"))
-        x #?(:clj       (-> public .getAffineXCoord .toBigInteger (.toString 16))
-             :cljs (-> public (gobj/get "x") .toString (.replace #"^0x" "") encodings/pad-hex))
-        y #?(:clj       (-> public .getAffineYCoord .toBigInteger (.toString 16))
-             :cljs (-> public (gobj/get "y") .toString (.replace #"^0x" "") encodings/pad-hex))
+  (let [private #?(:clj (:private pair)
+                   :cljs (gobj/get pair "private"))
+        ^ECPoint public #?(:clj (:public pair)
+                           :cljs (gobj/get pair "public"))
+        x #?(:clj  (-> public .getAffineXCoord .toBigInteger
+                       (.toString 16) (pad-to-length 64))
+             :cljs (-> public (gobj/get "x") .toString
+                       (.replace #"^0x" "") encodings/pad-hex))
```

Comment:
It's reassuring to see that the cljs has padding too - does it do something similar?

Reply:
Similar, yes, but pretty different in details. It only pads with zero or one leading zeroes, based on whether or not the hex value is odd (pads if yes, not if no).

I'd feel better if this were exactly what was needed here too, but I've seen both length 62 and 63, and they both fail in the same way (needs to be length 64 to work).

Reply:
I think 0-padding is a fine solution. I'd rather see it not fail with shorter values, smaller coordinate value is fine... so ideally it would accept 62-bit length. Regardless the zero-padding is perhaps the easiest one to implement with the least amount of code so good here.